src/util/persisted-data.ts

```diff
@@ -15,7 +15,7 @@ const jsonStringify = (data: any) => JSON.stringify(data, undefined, 2);
  */
 const migrateToJsonIfNeeded = async (migrationContext: IMigrationContext) => {
   const legacyFile = getLegacyRepoListFile(migrationContext);
-  if (await fs.existsAsync(legacyFile)) {
```

Comment:
existsAsync is deprecated for node 8+

Reply:
This is provided by `fs-extra-promise`, it isn't the version from the native `fs` module. I'd prefer to keep using async fs methods everywhere if possible!

Reply:
`fs-extra-promise` has fallen out of maintenance (see https://github.com/overlookmotel/fs-extra-promise#readme).  I'd prefer to find an alternative to `fs-extra-promise`.  

Perhaps use `fs.access` (see https://nodejs.org/api/fs.html#fs_fspromises_access_path_mode)?

Reply:
Node 8+ supports `fs.access`.

Reply:
Looks like `fs-extra` actually has native promise support! We should definitely move to that. If you wouldn't mind, could you back this change out of this PR so we can move the whole project to `fs-extra` in a separate PR?

Reply:
Will do :)